src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeMemory.Unix.cs

```diff
@@ -0,0 +1,158 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>This class contains methods that are mainly used to manage native memory.</summary>
+    public static unsafe class NativeMemory
+    {
+        /// <summary>Allocates an aligned block of memory of the specified size and alignment, in bytes.</summary>
+        /// <param name="byteCount">The size, in bytes, of the block to allocate.</param>
+        /// <param name="alignment">The alignment, in bytes, of the block to allocate. This must be a power of <c>2</c>.</param>
+        /// <returns>A pointer to the allocated aligned block of memory.</returns>
+        /// <exception cref="ArgumentException"><paramref name="alignment" /> is not a power of two.</exception>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory with <paramref name="alignment" /> failed.</exception>
+        /// <remarks>
+        ///     <para>This method allows <paramref name="byteCount" /> to be <c>0</c> and will return a valid pointer that should not be dereferenced and that should be passed to free to avoid memory leaks.</para>
+        ///     <para>This method is a thin wrapper over the C <c>aligned_alloc</c> API or a platform dependent aligned allocation API such as <c>_aligned_malloc</c> on Win32.</para>
+        ///     <para>This method is not compatible with <see cref="Free" /> or <see cref="Realloc" />, instead <see cref="AlignedFree" /> should be called.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* AlignedAlloc(nuint byteCount, nuint alignment)
+        {
+            if (!BitOperations.IsPow2(alignment))
+            {
+                // The C standard doesn't define what a valid alignment is, however Windows and POSIX requires a power of 2
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_AlignmentMustBePow2);
+            }
+
+            // The C standard and POSIX requires size to be a multiple of alignment and we want an "empty" allocation for zero
+            // POSIX additionally requires alignment to be at least sizeof(void*)
+
+            // The adjustment for byteCount can overflow here, but such overflow is "harmless". This is because of the requirement
+            // that alignment be a power of two and that byteCount be a multiple of alignment. Given both of these constraints
+            // we should only overflow for byteCount > (nuint.MaxValue & ~(alignment - 1)). When such an overflow occurs we will
+            // get a result that is less than alignment which will cause the allocation to fail.
+
+            void* result = Interop.Sys.AlignedAlloc(Math.Max(alignment, sizeof(void*)), (byteCount != 0) ? (byteCount + (alignment - 1)) & ~(alignment - 1) : alignment);
```

Comment:
Does the formula for the second argument need to use the alignment after it has been bumped up to `sizeof(void*)`?

Reply:
Yes, it does. Thanks for the catch.

Reply:
Should be resolved now.